tests/plugins/find_and_replace.test.ts

```diff
@@ -32,62 +32,135 @@ let searchOptions: SearchOptions;
+    expect(model.getters.getSelection().zones).toEqual([toZone("A2")]);
   });
 
-  test("change the search", async () => {
+  test("change the search for activeSheet searchScope", () => {
     model.dispatch("UPDATE_SEARCH", { toSearch: "hello", searchOptions });
     model.dispatch("SELECT_SEARCH_NEXT_MATCH");
     let matches = model.getters.getSearchMatches();
     let matchIndex = model.getters.getCurrentSelectedMatchIndex();
     expect(matches).toHaveLength(2);
     expect(matchIndex).toStrictEqual(1);
-    expect(matches[0]).toStrictEqual({ col: 0, row: 0, selected: false });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 1, selected: true });
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 0, selected: false });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: true });
+    model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
+    matches = model.getters.getSearchMatches();
+    matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(3);
+    expect(matchIndex).toStrictEqual(0);
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ sheetId: "s1", col: 0, row: 3, selected: false });
+  });
 
+  test("change the search for allSheet searchScope", () => {
+    searchOptions.searchScope = "allSheets";
+    model.dispatch("UPDATE_SEARCH", { toSearch: "hello", searchOptions });
+    model.dispatch("SELECT_SEARCH_NEXT_MATCH");
+    let matches = model.getters.getSearchMatches();
+    let matchIndex = model.getters.getCurrentSelectedMatchIndex();
+    expect(matches).toHaveLength(2);
+    expect(matchIndex).toStrictEqual(1);
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 0, selected: false });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: true });
     model.dispatch("UPDATE_SEARCH", { toSearch: "1", searchOptions });
     matches = model.getters.getSearchMatches();
     matchIndex = model.getters.getCurrentSelectedMatchIndex();
     expect(matches).toHaveLength(4);
     expect(matchIndex).toStrictEqual(0);
-    expect(matches[0]).toStrictEqual({ col: 0, row: 1, selected: true });
-    expect(matches[1]).toStrictEqual({ col: 0, row: 2, selected: false });
-    expect(matches[2]).toStrictEqual({ col: 0, row: 3, selected: false });
-    expect(matches[3]).toStrictEqual({ col: 0, row: 4, selected: false });
+    expect(matches[0]).toStrictEqual({ sheetId: "s1", col: 0, row: 1, selected: true });
+    expect(matches[1]).toStrictEqual({ sheetId: "s1", col: 0, row: 2, selected: false });
+    expect(matches[2]).toStrictEqual({ sheetId: "s1", col: 0, row: 3, selected: false });
+    expect(matches[3]).toStrictEqual({ sheetId: "s2", col: 0, row: 1, selected: false });
```

Comment:
Yeah there's way too much stuff that we don't care about in the beforeEach, we dont need to test 3-4 matches in each and every test...